mlir/include/mlir/IR/OpDefinition.h

```diff
@@ -1309,6 +1311,25 @@ struct HasParent {
       return llvm::cast<ParentOpType>(parent);
     }
   };
+
+private:
+  /// A class is an op interface if it has a `getInterfaceName` function.
```

Comment:
Ops and op interfaces inherit from `OpState`. If you know a better way of checking if a class is an interface, let me know. I also thought about calling the function `getOperationName` instead of `getInterfaceName`, but that could be confusing.

Reply:
We actually have logic for this already at https://github.com/llvm/llvm-project/blob/50b45b24220ead33cf5cedc49c13e0336297e4eb/mlir/include/mlir/Support/InterfaceSupport.h#L270 which is used to make `DenseMap` work with interfaces: https://github.com/llvm/llvm-project/blob/50b45b24220ead33cf5cedc49c13e0336297e4eb/mlir/include/mlir/IR/OpDefinition.h#L2130